internal/controller/replication.storage/status.go

```diff
@@ -76,24 +83,28 @@ func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGenerat
 // sets conditions when volume promotion was failed due to failed validation.
 func setFailedValidationCondition(conditions *[]metav1.Condition, observedGeneration int64) {
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            "failed to promote",
```

Comment:
This message is not helpful - why did it fail? what is the the way to recover from the error?

In ramen we want to propagate the errors to the ramen resources conditions. If we include good error message in the ramen resource debugging issues will be come much easier, since the error messages would propagate to the DRPC on the hub.

But in this case the error is actually wrong - since the issue is failing initial validation. We did not try to promote and did not fail to promote since promotion is not possible.

Maybe use: "Promotion not possible due to failed precondition"

Reply:
yes, this condition gets excuted when the precondition fails due to initial validation and as mentioned earlier, the reason for error will be reflected in status.message. yes we can use the above msg, i will update it thanks